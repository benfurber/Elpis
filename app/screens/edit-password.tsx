import React, { Component } from "react";
import { ScrollView, StyleSheet } from "react-native";
import { withMappedNavigationParams } from "react-navigation-props-mapper";

import { ButtonClose, BackgroundContainer, FormAddPassword } from "components";
import { NavigationType } from "interfaces";
import { colours, layout } from "styles";

interface Props {
  navigation: NavigationType;
  onPress: () => void;
}

class EditPasswordScreen extends Component<Props> {
  render() {
    const { navigation } = this.props;

    return (
      <BackgroundContainer header={<ButtonClose navigation={navigation} />}>
        <ScrollView style={styles.scrollView}>
          <FormAddPassword
            navigation={navigation}
            onPress={() => this.props.onPress()}
          />
        </ScrollView>
      </BackgroundContainer>
    );
  }
}

const styles = StyleSheet.create({
  body: {
    width: "100%",
  },
  scrollView: {
    backgroundColor: colours.whiteTransparent,
    borderTopLeftRadius: layout.borderRadiusL,
    borderTopRightRadius: layout.borderRadiusL,
    height: "100%",
    padding: layout.spacingL,
  },
});

const wrappedEditPasswordScreen = withMappedNavigationParams()(
  EditPasswordScreen,
);
export {
  wrappedEditPasswordScreen as EditPasswordScreen,
  EditPasswordScreen as UnwrappedEditPasswordScreen,
};
