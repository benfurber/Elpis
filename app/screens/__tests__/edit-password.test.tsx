import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { UnwrappedEditPasswordScreen } from "../edit-password";

let navigation;
jest.mock(navigation, () => jest.fn());

describe("UnwrappedEditPasswordScreen", () => {
  it("renders correctly", () => {
    const component = shallow(
      <UnwrappedEditPasswordScreen
        navigation={navigation}
        onPress={() => jest.fn()}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
