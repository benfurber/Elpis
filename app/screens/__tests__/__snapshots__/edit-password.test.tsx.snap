// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnwrappedEditPasswordScreen renders correctly 1`] = `
<BackgroundContainer
  header={<ButtonClose />}
>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.85)",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": "100%",
        "padding": 20,
      }
    }
  >
    <FormAddPassword
      onPress={[Function]}
    />
  </ScrollViewMock>
</BackgroundContainer>
`;
